moveit_ros/moveit_servo/src/collision_check.cpp

```diff
@@ -64,6 +64,7 @@ CollisionCheck::CollisionCheck(ros::NodeHandle& nh, const moveit_servo::ServoPar
   // Init collision request
   collision_request_.group_name = parameters_.move_group_name;
   collision_request_.distance = true;  // enable distance-based collision checking
+  collision_request_.contacts = true;  // Record the names of collision pairs
```

Comment:
I'm not sure if this slows down collision checking or not... It might be worth profiling this method to see

Reply:
You may also want to add a rosparam for optionally storing and logging the collision contacts.

Reply:
I feel like @mamoll might now if storing all of the collision pairs slows down collision checking (versus potentially returning after the collision checker finds the first collision)

Reply:
I think @mlautman's intuition is right. Definitely something to check. I see that you already ask for distance information. This adds significant computational cost. The difference with just a boolean collision check can be orders of magnitude. The code path in FCL is entirely different depending on whether you want a collision check or distance info. Not sure how much worse it gets if you add contact info on top of that.

Reply:
I dropped a high-resolution timer in the CC loop and checked runtime, with and without contacts recorded. The difference is really small.

with contacts recorded:  0.017-0.026s per loop
without contacts recorded: 0.016-0.028s

I think there would be a bigger difference if the robot were in contact, but it would have stopped moving by then anyway. So we don't care about that.

Reply:
I'm not sure I understand why we wouldn't care about run time if the robot is in contact. Maybe I am not quite understanding how this feature will be used. 

My assumption is that this would be used to check if a joint motion would result in a collision before sending a joint command and wait until the user sends a command that isn't in collision. If that is the case, wouldn't we still want the cycle time to be really fast so that it can keep up with new commands at a high rate? Is the assumption that even if it takes longer in a cycle when the joint command would result in a collision, that eventually a new command would come in with a non-colliding command and the cycle time would be really fast again. This seems non-deterministic to me but I am also not overly familiar with the majority of the code base here.

Reply:
What you're describing would be nice but it doesn't work like that. The jogger just stops if it's in collision. There's an optional buffer so it will start decelerating before it collides.

So yeah, once it collides it's stuck

Reply:
Also, collision checking is pretty slow (about 50 Hz). Usually robot commands will go out at a higher rate than that

Reply:
Okay, this sounds reasonable. I still don't fully understand how the collision checking and the jogging are run at different rates but I trust that the design decision you are making here are sound. If you are confident that this won't affect performance and that there isn't much need for a parameter that would disable verbose collision checking than I'm happy to merge.